File: tailwind-admin-reactjs-free/package/src/types/asistencia.ts
export type EstadoAsistencia = 'Presente' | 'Ausente' | 'Tardia' | 'Justificado';

export interface AsistenciaRosterItem {
  id_estudiante: number;
  name_estudiante: string;
  sec_name_estudiante: string;
  estado: EstadoAsistencia | null;
  observacion: string | null;
}

export interface AsistenciaRoster {
  id_grupo: number;
  fecha: string;
  registros: AsistenciaRosterItem[];
}

export interface AsistenciaRegistroIn {
  id_estudiante: number;
  estado: EstadoAsistencia;
  observacion?: string | null;
}

export interface AsistenciaBatchIn {
  id_grupo: number;
  fecha: string;
  registros: AsistenciaRegistroIn[];
}

File: backend/app/schemas/asistencia.py
from datetime import date
from typing import Literal

from pydantic import BaseModel, Field

EstadoAsistencia = Literal["Presente", "Ausente", "Tardia", "Justificado"]


class AsistenciaRegistroIn(BaseModel):
    id_estudiante: int
    estado: EstadoAsistencia
    observacion: str | None = Field(default=None, max_length=255)


class AsistenciaBatchIn(BaseModel):
    id_grupo: int
    fecha: date
    registros: list[AsistenciaRegistroIn]


class AsistenciaRosterItem(BaseModel):
    id_estudiante: int
    name_estudiante: str
    sec_name_estudiante: str
    estado: EstadoAsistencia | None = None
    observacion: str | None = None


class AsistenciaRosterOut(BaseModel):
    id_grupo: int
    fecha: date
    registros: list[AsistenciaRosterItem]
